include/albatross/src/utils/vector_utils.hpp

```diff
@@ -31,6 +31,30 @@ inline std::vector<double> linspace(double a, double b, std::size_t n) {
   return xs;
 }
 
+inline bool all(const std::vector<bool> &xs) {
+  if (xs.size() == 0) {
+    return false;
```

Comment:
I've usually seen this base case for this all type function default to `true` when empty and the `any` function default to `false` when empty. I guess in this situation you're logic would be simpler with both returning `false`?

Reply:
Yeah I wasn't totally sure about the empty cases.  I think I agree that `all` should probably return `true`.  I was tempted to put an assert in there ... but that didn't feel right.